builder/views.py

```diff
@@ -45,7 +46,10 @@ def _handle_post(request, draft):
     elif action == "discard":
         messages.add_message(request, messages.INFO, "Your changes have been discarded")
         actions.discard_draft(draft=draft)
-        return redirect(draft.codelist)
+        # check if the discarded draft's codelist still exists (i.e. this was not the only version)
+        if draft.codelist.id is not None:
```

Comment:
Ooh, I'd not thought of checking the ID before!  Could you optimise this further by using `draft.codelist_id` or would that not have been updated because it's on the now-deleted Draft instance?

Reply:
Hmm.  I don't know - I'll try and see

Reply:
Nope, apparently you can't.  `codelist_id` is still not None at this point.